File: EpersBackend/Epers.Models/Obiectiv/Obiective.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Epers.Models.Obiectiv
{
    public class Obiective
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }

        public string Denumire { get; set; } = string.Empty;

        public string IdAngajat { get; set; } = string.Empty;

        public string MatricolaAngajat { get; set; } = string.Empty;

        public string IdSuperior { get; set; } = string.Empty;

        public string MatricolaSuperior { get; set; } = string.Empty;

        public int? IdCompartiment { get; set; }

        public int? IdPost { get; set; }

        public DateTime? DataIn { get; set; }

        public DateTime? DataSf { get; set; }

        public decimal? ValMin { get; set; }

        public decimal? ValTarget { get; set; }

        public decimal? ValMax { get; set; }

        public decimal? BonusMin { get; set; }

        public decimal? BonusTarget { get; set; }

        public decimal? BonusMax { get; set; }

        public string Frecventa { get; set; } = string.Empty;

        public string Tip { get; set; } = string.Empty;

        public string ValoareRealizata { get; set; } = string.Empty;

        public bool? IsRealizat { get; set; }

        public decimal? Pondere { get; set; }

        public bool? IsActive { get; set; }

        public bool IsFaraProcent { get; set; }

        public bool? IsBonusProcentual { get; set; }

        public string UpdateId { get; set; } = string.Empty;

        public DateTime? UpdateDate { get; set; }

        public int? IdFirma { get; set; }
        
        public bool? IsInDesfasurare { get; set; }
    }
}

File: epers-front-end/src/app/models/obiective/Obiective.ts
import { SubalterniDropdown } from "../useri/User";

export interface ObiectiveListaSubalterniDisplayModel {
    listaSubalterni: SubalterniDropdown[];
    pages: number;
    currentPage: number;
}

export interface ObiectivTemplate {
    id: number;
    denumire: string;
    valMin: number | null;
    valTarget: number | null;
    valMax: number | null;
    bonusMin: number | null;
    bonusTarget: number | null;
    bonusMax: number | null;
    isFaraProcent: boolean;
    tip: string;
    dataIn: Date | null;
    dataSf: Date | null;
    frecventa: string;
    isBonusProcentual: boolean | null;
}

export interface Obiective {
    id: number;
    denumire: string;
    idAngajat: string;
    matricolaAngajat: string;
    idSuperior: string;
    matricolaSuperior: string;
    idCompartiment: number | null;
    idPost: number | null;
    dataIn: Date | null;
    dataSf: Date | null;
    valMin: number | null;
    valTarget: number | null;
    valMax: number | null;
    bonusMin: number | null;
    bonusTarget: number | null;
    bonusMax: number | null;
    frecventa: string;
    tip: string;
    valoareRealizata: string;
    isRealizat: boolean;
    pondere: number | null;
    isActive: boolean | null;
    isFaraProcent: boolean;
    updateId: string;
    updateDate: Date | null;
    isBonusProcentual: boolean | null;
}

export interface ObiectiveDisplayModel {
    listaObActuale: Obiective[];
    pages: number;
    currentPage: number;
}
